// expo-cli/build/commands/eas-build/credentialsSync/action.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = credentialsSyncAction;
exports.updateLocalCredentialsAsync = updateLocalCredentialsAsync;

function _CommandError() {
  const data = _interopRequireDefault(require("../../../CommandError"));

  _CommandError = function () {
    return data;
  };

  return data;
}

function _context() {
  const data = require("../../../credentials/context");

  _context = function () {
    return data;
  };

  return data;
}

function credentialsJsonUpdateUtils() {
  const data = _interopRequireWildcard(require("../../../credentials/credentialsJson/update"));

  credentialsJsonUpdateUtils = function () {
    return data;
  };

  return data;
}

function _route() {
  const data = require("../../../credentials/route");

  _route = function () {
    return data;
  };

  return data;
}

function _SetupAndroidKeystore() {
  const data = require("../../../credentials/views/SetupAndroidKeystore");

  _SetupAndroidKeystore = function () {
    return data;
  };

  return data;
}

function _SetupIosBuildCredentials() {
  const data = require("../../../credentials/views/SetupIosBuildCredentials");

  _SetupIosBuildCredentials = function () {
    return data;
  };

  return data;
}

function _log() {
  const data = _interopRequireDefault(require("../../../log"));

  _log = function () {
    return data;
  };

  return data;
}

function _prompts() {
  const data = _interopRequireDefault(require("../../../prompts"));

  _prompts = function () {
    return data;
  };

  return data;
}

function _ios() {
  const data = require("../build/utils/ios");

  _ios = function () {
    return data;
  };

  return data;
}

function _types() {
  const data = require("../types");

  _types = function () {
    return data;
  };

  return data;
}

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

async function credentialsSyncAction(projectDir, options) {
  if (options.parent.nonInteractive) {
    throw new (_CommandError().default)('This command is not supported in --non-interactive mode');
  }

  const {
    update,
    platform
  } = await (0, _prompts().default)([{
    type: 'select',
    name: 'update',
    message: 'What do you want to do?',
    choices: [{
      title: 'Update credentials on the Expo servers with the local credentials.json contents',
      value: 'remote'
    }, {
      title: 'Update or create local credentials.json with credentials from the Expo servers',
      value: 'local'
    }]
  }, {
    type: 'select',
    name: 'platform',
    message: 'Which platform would you like to update?',
    choices: [{
      title: 'Android',
      value: _types().BuildCommandPlatform.ANDROID
    }, {
      title: 'iOS',
      value: _types().BuildCommandPlatform.IOS
    }, {
      title: 'both',
      value: _types().BuildCommandPlatform.ALL
    }]
  }]);

  if (update === 'local') {
    await updateLocalCredentialsAsync(projectDir, platform);
  } else {
    await updateRemoteCredentialsAsync(projectDir, platform);
  }
}

async function updateRemoteCredentialsAsync(projectDir, platform) {
  const ctx = new (_context().Context)();
  await ctx.init(projectDir);

  if (!ctx.hasProjectContext) {
    throw new Error('project context is required'); // should be checked earlier
  }

  if ([_types().BuildCommandPlatform.ALL, _types().BuildCommandPlatform.ANDROID].includes(platform)) {
    const experienceName = `@${ctx.manifest.owner || ctx.user.username}/${ctx.manifest.slug}`;
    await (0, _route().runCredentialsManager)(ctx, new (_SetupAndroidKeystore().SetupAndroidBuildCredentialsFromLocal)(experienceName));
  }

  if ([_types().BuildCommandPlatform.ALL, _types().BuildCommandPlatform.IOS].includes(platform)) {
    var _ctx$manifest$owner;

    const bundleIdentifier = await (0, _ios().getBundleIdentifier)(projectDir, ctx.manifest);
    const appLookupParams = {
      accountName: (_ctx$manifest$owner = ctx.manifest.owner) !== null && _ctx$manifest$owner !== void 0 ? _ctx$manifest$owner : ctx.user.username,
      projectName: ctx.manifest.slug,
      bundleIdentifier
    };
    await (0, _route().runCredentialsManager)(ctx, new (_SetupIosBuildCredentials().SetupIosBuildCredentialsFromLocal)(appLookupParams));
  }
}

async function updateLocalCredentialsAsync(projectDir, platform) {
  const ctx = new (_context().Context)();
  await ctx.init(projectDir);

  if (!ctx.hasProjectContext) {
    throw new Error('project context is required'); // should be checked earlier
  }

  if ([_types().BuildCommandPlatform.ALL, _types().BuildCommandPlatform.ANDROID].includes(platform)) {
    (0, _log().default)('Updating Android credentials in credentials.json');
    await credentialsJsonUpdateUtils().updateAndroidCredentialsAsync(ctx);
  }

  if ([_types().BuildCommandPlatform.ALL, _types().BuildCommandPlatform.IOS].includes(platform)) {
    const bundleIdentifier = await (0, _ios().getBundleIdentifier)(projectDir, ctx.manifest);
    (0, _log().default)('Updating iOS credentials in credentials.json');
    await credentialsJsonUpdateUtils().updateIosCredentialsAsync(ctx, bundleIdentifier);
  }
}
